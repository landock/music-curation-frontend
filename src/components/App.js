import React, { Component } from 'react';
import './App.css';

export default class App extends Component {
  render() {
    const { curations } = this.props;
    let id, imageUrl, tracks, curationName;
    if (curations && curations[0]) {
      id = curations[0].id;
      imageUrl = curations[0].imageUrl;
      tracks = curations[0].tracks;
      curationName = curations[0].name;
    }
    console.log(this.props);
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Welcome to Music Curation Apppppp</h1>
        </header>
        <h1>
          {`${curationName} - ${id}`}
          <img style={{ height: '120px' }} src={imageUrl} />
        </h1>
        {tracks &&
          tracks.map(track => {
            return (
              <div key={track.id}>
                <img style={{ height: '120px' }} src={track.imageUrl} />
                <p>{`${track.artistName} - ${track.trackName}`}</p>
              </div>
            );
          })}
      </div>
    );
  }
}
