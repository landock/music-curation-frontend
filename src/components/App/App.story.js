import React from 'react';

import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import thunk from 'redux-thunk';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import App from './';

const curations = [
  {
    id: '23',
    imageUrl: 'https://placeimg.com/250/250/tech/grayscale',
    tracks: [
      {
        id: '12',
        imageUrl: 'https://placeimg.com/250/250/animals',
        artistName: 'spongebop',
        trackName: 'squarewave',
      },
    ],
    name: 'story testing',
  },
];

const mockStore = configureMockStore([thunk]);
const store = mockStore({ Curations: curations });

storiesOf('App', module)
  .addDecorator(story => {
    return <Provider store={store}>{story()}</Provider>;
  })
  .add('default', () => <App curations={curations} />);
storiesOf('App', module)
  .addDecorator(story => {
    return <Provider store={store}>{story()}</Provider>;
  })
  .add('no props', () => <App curations={null} />);
