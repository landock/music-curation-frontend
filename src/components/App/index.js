import React, { Component } from 'react';
import './App.css';
import logo from '../../img/logo.svg';
import CurationsContainer from '../../containers/CurationsContainer';
import TagCollection from '../TagCollection';

export default class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <div style={{ width: '100%', height: '50px', background: '#fff' }}>
            <img width="100" src={logo} />
          </div>
          <div
            style={{
              backgroundColor: '#354244',
              height: '100px',
              display: 'flex',
              alignItems: 'center',
              justifyContent: 'center',
            }}
          >
            <input
              style={{
                width: '50%',
                padding: '5px',
                height: '25px',
                fontSize: '50%',
              }}
              type="text"
              placeholder="Search Curations"
            />
          </div>
          <div style={{ background: 'black', height: '150px', width: '100%' }}>
            <div>
              <label>Environment</label>
              <select />
            </div>
            <div>
              <label>Status</label>
              <select />
            </div>
            <div>
              <span>Tags </span>
              <TagCollection
                tags={[{ id: 1, title: 'what' }, { id: 2, title: 'when' }]}
              />
            </div>
          </div>
        </header>
        <CurationsContainer />
      </div>
    );
  }
}
