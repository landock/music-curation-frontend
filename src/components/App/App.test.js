import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';

import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';

import App from './index';

describe('App Component', () => {
  let component, mockStore, store;
  beforeEach(() => {
    component = shallow(<App />);
    mockStore = configureMockStore([thunk]);
  });
  it('renders without crashing', () => {
    let store = mockStore({});
    let tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <App />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render shit if curations prop exists', () => {
    let name = 'Test fuckin Name';
    let id = '11';
    let curationMarkup = (
      <p>
        {name} - {id}
      </p>
    );
    let artistName = 'George';
    let trackName = 'Ullmer';
    let trackMarkup = (
      <p>
        {artistName} - {trackName}
      </p>
    );

    component.setProps({
      curations: [
        {
          id,
          name,
          imageUrl: 'test',
          tracks: [
            {
              id: 22,
              artistName,
              trackName,
              imageUrl: 'test',
            },
          ],
        },
      ],
    });

    expect(component.contains(curationMarkup));
    expect(component.contains(trackMarkup));
  });
});
