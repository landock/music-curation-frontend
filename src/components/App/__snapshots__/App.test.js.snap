// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      style={
        Object {
          "background": "#fff",
          "height": "50px",
          "width": "100%",
        }
      }
    >
      <img
        src="logo.svg"
        width="100"
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#354244",
          "display": "flex",
          "height": "100px",
          "justifyContent": "center",
        }
      }
    >
      <input
        placeholder="Search Curations"
        style={
          Object {
            "fontSize": "50%",
            "height": "25px",
            "padding": "5px",
            "width": "50%",
          }
        }
        type="text"
      />
    </div>
    <div
      style={
        Object {
          "background": "black",
          "height": "150px",
          "width": "100%",
        }
      }
    >
      <div>
        <label>
          Environment
        </label>
        <select />
      </div>
      <div>
        <label>
          Status
        </label>
        <select />
      </div>
      <div>
        <span>
          Tags 
        </span>
        <span
          style={
            Object {
              "background": "red",
              "color": "white",
              "display": "inline-block",
              "margin": "5px",
              "padding": "5px",
            }
          }
        >
          what
        </span>
        <span
          style={
            Object {
              "background": "red",
              "color": "white",
              "display": "inline-block",
              "margin": "5px",
              "padding": "5px",
            }
          }
        >
          when
        </span>
      </div>
    </div>
  </header>
  <p>
    No Curations Found
  </p>
</div>
`;
